### PR TITLE
[MINI-3240] add user point error 

### PR DESCRIPTION
**Pull request by khairul-alam-licon**
# Description
Primary work of user's point interface has been added by https://github.com/rakutentech/android-miniapp/pull/331.
This PR aims to add `MiniAppPointsError` for sending it to the mini-apps.

## Links
- MINI-3240

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Review by corycaywood — APPROVED** [review]
👍

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/errors/MiniAppPointsError.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    MiniAppBridgeError(type, message) {
+
+    companion object {
+        /**
+         *  send custom error message from host app.
+         *  @property message error message send to mini app.
+         */
+        fun custom(message: String) = MiniAppPointsError(message = message)
+    }
+}
```
Comment (munir-rakuten) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/errors/MiniAppPointsError.kt:16]: Do we need this class? As we don't have any specific type for this error and only message, so may be we can use `MiniAppBridgeError` for this as both looks identical..but if we include specific types in future we can implement like this, what do you think?

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt [hunk 2]
```diff
@@ -222,11 +223,11 @@ class MiniAppDisplayActivity : BaseActivity() {
 
             override fun getPoints(
                 onSuccess: (points: Points) -> Unit,
-                onError: (message: String) -> Unit
+                onError: (pointsError: MiniAppPointsError) -> Unit
             ) {
                 val points = AppSettings.instance.points
                 if (points != null) onSuccess(points)
-                else onError("There is no points found in HostApp.")
+                else onError(MiniAppPointsError(null, "There is no points found in HostApp."))
```
Comment (munir-rakuten) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:230]: We can use `MiniAppPointsError.custom(""There is no points found in HostApp.")` like this? or make the type nullable?

**Review by munir-rakuten — COMMENTED** [review]
Looks good, I placed some suggestions though..

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/errors/MiniAppPointsError.kt (hunk 1 above)
Comment (khairul-alam-licon) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/errors/MiniAppPointsError.kt:16]: Yeah, we can use `MiniAppPointsError` to denote the feature specific error class, also we can use the `custom` function to pass the message only.

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt (hunk 2 above)
Comment (khairul-alam-licon) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt]: Updated.

**Review by munir-rakuten — APPROVED** [review]

**Pull request merged**